### PR TITLE
Some pre-release tasks

### PR DESCRIPTION
**Pull request by gassmoeller**
Just going through some of our pre release tasks. I also updated the release checklist with a bit more instructions, and reordered it to separate tasks that can be done before the release from tasks that are directly related to the release process.

### CODE REVIEW

File: include/aspect/boundary_temperature/dynamic_core.h [hunk 1]
```diff
@@ -109,8 +109,8 @@ namespace aspect
          * at the inner and outer boundaries.
          *
          * @param boundary_indicator The boundary indicator of the part of the boundary
-         *   of the domain on which the point is located at which we are requesting the
```
Comment (tjhei) [inline include/aspect/boundary_temperature/dynamic_core.h:112]: I normally go through the diffs manually and only apply it in wrong places but not in locations like here where the indentation makes sense. Thoughts?

**Comment by tjhei** [comment]
Would you mind splitting into several PRs? The size of changes makes it hard to review (chrome is lagging).

**Comment by gassmoeller** [comment]
Sure, sorry for the mess in this PR. This has been split into the other PRs referencing this one (6085-6088). About the indentation: Yes, I think there are several options which indentation changes to accept or not accept. I was more conservative now in #6088 and kept more manual indentation. I just find it a bit annoying to go through a hundred chunks by hand and decide which ones to keep. Maybe we can come up with a better process?

**Comment by tjhei** [comment]
> I just find it a bit annoying to go through a hundred chunks by hand and decide which ones to keep.

Yeah, it is. I am happy to do the task for now, if you want.

> Maybe we can come up with a better process?

It is difficult to fix the script to detect all cases of intentional indentation. I guess one could teach it about @deprecated and @param, though.

**Pull request closed without merging**